ejb3/src/main/java/org/jboss/as/ejb3/remote/EJBRemoteTransactionPropagatingInterceptor.java

```diff
@@ -114,7 +114,7 @@ private void createOrResumeXidTransaction(final XidTransactionID xidTransactionI
         } else {
             // begin a new tx and add it to the tx repository
             // TODO: Fix the tx timeout to accept a value from the client (WFLY-2789)
-            final Transaction newSubOrdinateTx = this.ejbRemoteTransactionsRepository.importTransaction(xidTransactionID, Integer.MAX_VALUE);
+            final Transaction newSubOrdinateTx = this.ejbRemoteTransactionsRepository.importTransaction(xidTransactionID, Integer.parseInt(System.getProperty("remote-tx-timeout", "31536000")));
```

Comment:
My interpretation of the discussion on this is that WFLY-2789 is the suggested long term fix for this, which AIUI is not meant to involve any server side setting. In that case there may very well never be a proper management attribute for this. If I am correct this fix really is a workaround and may never be formalized into the server management API.

The above is not meant to argue against this fix; it's meant to argue against assumptions as to what will happen in the future.